module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -216,16 +221,11 @@ kv_alloc(spl_kmem_cache_t *skc, int size, int flags)
 		 *
 		 * For non-GFP_KERNEL allocations we stick to __vmalloc.
 		 */
-		if ((lflags & GFP_KERNEL) == GFP_KERNEL) {
-			ptr = spl_kvmalloc(size, lflags);
-		} else {
-			ptr = __vmalloc(size, lflags | __GFP_HIGHMEM,
-			    PAGE_KERNEL);
-		}
+		ptr = spl_kvmalloc(size, lflags);
+	} else {
+		ptr = __vmalloc(size, lflags | __GFP_HIGHMEM, PAGE_KERNEL);
 	}
 
-	/* Resulting allocated memory will be page aligned */
-	ASSERT(IS_P2ALIGNED(ptr, PAGE_SIZE));
```

Comment:
The matching `ASSERT(IS_P2ALIGNED(ptr, PAGE_SIZE));` in `kv_free()` would also need to be removed.

Reply:
oops.. looks like I forgot to commit and push that o.O